barretenberg/cpp/src/barretenberg/ultra_honk/ultra_verifier.cpp

```diff
@@ -22,6 +23,45 @@ template <typename Flavor> bool UltraVerifier_<Flavor>::verify_proof(const HonkP
             transcript->template get_challenge<FF>("Sumcheck:gate_challenge_" + std::to_string(idx)));
     }
 
+    const auto recover_fq_from_public_inputs = [](std::array<FF, 4> limbs) {
```

Comment:
parses out the IPA claim from the public inputs using the vkey and runs the. native verifier